### ISSUE TITLE
[Request] Hardware Encryption Support

### ISSUE THREAD

**Issue by UtilFunction** [archlinux/archinstall#2589]
### describe the request

Cryptsetup has been supporting hardware encryption with TCG Opal-compliant drives since [version 2.7.0](https://mirrors.edge.kernel.org/pub/linux/utils/cryptsetup/v2.7/v2.7.0-ReleaseNotes). The forthcoming [version 41](https://www.phoronix.com/news/Fedora-41-Self-Encrypting-Drive) of the Fedora installer will also support this functionality. It is possible to utilise either hardware encryption only, or in conjunction with software encryption by incorporating a straightforward flag.

It would be nice to have this option within archinstall.

**Comment by Torxed** [comment]
Would this be something different to the HSM support we've had since [v2.5.0](https://github.com/archlinux/archinstall/releases/tag/v2.5.0)?

**Comment by UtilFunction** [comment]
Yes, these two are not related. HSMs basically only deal with authentication and the handling of keys. Self encrypting drives perform encryption autonomously and hence without overhead on the host's CPU [which is considerable even with AES-NI](https://scs.community/2023/02/24/impact-of-disk-encryption/).
